pkg/sfu/publisher.go

```diff
@@ -84,9 +87,14 @@ func NewPublisher(id string, session Session, cfg *WebRTCTransportConfig) (*Publ
 			p.mu.Unlock()
 		} else {
 			p.mu.Lock()
-			p.tracks = append(p.tracks, publisherTracks{track, r, false})
+			publisherTrack = PublisherTrack{track, r, false}
+			p.tracks = append(p.tracks, publisherTrack)
 			p.mu.Unlock()
 		}
+
+		if p.onPublisherTrack != nil {
```

Comment:
you should call the method only when pub is true otherwise on simulcast you will fire the event up to 3 times.